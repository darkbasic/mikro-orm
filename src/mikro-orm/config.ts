import {Options} from '@mikro-orm/core';
import {PostgreSqlDriver} from '@mikro-orm/postgresql';
import {TsMorphMetadataProvider} from '@mikro-orm/reflection';

export const useDocker =
  process.env.DOCKER === 'true' ||
  process.env.DOCKER === 'TRUE' ||
  process.env.DOCKER === '1';

export const dbName = useDocker ? 'docker' : 'test';
const port = useDocker ? 5433 : 5432;
const user = useDocker ? 'postgres' : 'test';
const password = useDocker ? 'postgres' : 'test';

export const config: Options<PostgreSqlDriver> = {
  entities: ['./build/src/mikro-orm/entities'],
  entitiesTs: ['./src/mikro-orm/entities'],
  metadataProvider: TsMorphMetadataProvider,
  dbName,
  port,
  user,
  password,
  type: 'postgresql',
  forceUtcTimezone: true,
  debug: true,
};
