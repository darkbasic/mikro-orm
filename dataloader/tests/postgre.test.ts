import {MikroORM} from 'mikro-orm';
import {PostgreSqlDriver} from 'mikro-orm/dist/drivers/PostgreSqlDriver';
import {
  initORMPostgreSql,
  wipeDatabasePostgreSql,
  createApolloServer,
  dropSchema,
} from './bootstrap';
import {ApolloServer, gql} from 'apollo-server';
import {createTestClient} from 'apollo-server-testing';

describe('Backend', () => {
  jest.setTimeout(10e3);
  let orm: MikroORM<PostgreSqlDriver>;
  let server: ApolloServer;

  const getSportsQuery = gql`
    {
      sports {
        id
        name
        specialties {
          id
          name
        }
      }
    }
  `;

  const getSpecialtiesQuery = gql`
    {
      specialties {
        id
        name
        sport {
          id
          name
        }
      }
    }
  `;

  const bigQuery = gql`
    {
      specialties {
        id
        name
        sport {
          id
          name
          sites {
            id
            name
            sports {
              id
              name
            }
            matches {
              id
              date
            }
          }
          specialties {
            id
            name
          }
        }
        matches {
          id
          date
          specialty {
            id
            name
          }
          site {
            id
            name
          }
          partecipants {
            id
            name
            createdMatches {
              id
              date
            }
            partecipatedMatches {
              id
              date
            }
          }
        }
      }
    }
  `;

  beforeAll(async () => (orm = await initORMPostgreSql()));
  beforeEach(async () => await wipeDatabasePostgreSql(orm.em));

  test('isConnected()', async () => {
    await expect(orm.isConnected()).resolves.toBe(true);
    await orm.close(true);
    await expect(orm.isConnected()).resolves.toBe(false);
    await orm.connect();
    await expect(orm.isConnected()).resolves.toBe(true);
  });

  it('should fetch all users', async () => {
    server = createApolloServer(orm);
    const {query} = createTestClient(server);
    const res = await query({
      query: gql`
        {
          users {
            id
            name
          }
        }
      `,
    });
    expect(res.data).toBeDefined();
    expect(res.errors).toBeUndefined();
    expect(res.data).toMatchSnapshot();
  });

  it('should fetch all sports and their specialties (One To Many)', async () => {
    server = createApolloServer(orm);
    const {query} = createTestClient(server);
    const res = await query({query: getSportsQuery});
    expect(res.data).toBeDefined();
    expect(res.errors).toBeUndefined();
    expect(res.data).toMatchSnapshot();
  });

  it('should return the same entities with EntityDataLoader', async () => {
    server = createApolloServer(orm);
    let {query} = createTestClient(server);
    const res = await query({query: getSportsQuery});
    server = createApolloServer(orm, false);
    query = createTestClient(server).query;
    const resDataloader = await query({query: getSportsQuery});
    expect(res.data).toBeDefined();
    expect(res.errors).toBeUndefined();
    expect(resDataloader.data).toBeDefined();
    expect(resDataloader.errors).toBeUndefined();
    expect(JSON.stringify(resDataloader.data)).toBe(JSON.stringify(res.data));
  });

  it('should be faster with EntityDataLoader', async () => {
    server = createApolloServer(orm);
    let {query} = createTestClient(server);
    const start = new Date();
    await query({query: getSportsQuery});
    const end = new Date();
    const diff = end.getTime() - start.getTime();
    server = createApolloServer(orm, false);
    query = createTestClient(server).query;
    const startDataloader = new Date();
    await query({query: getSportsQuery});
    const endDataloader = new Date();
    const diffDataloader = endDataloader.getTime() - startDataloader.getTime();
    console.log(`W/o dataloader: ${diff} ms`);
    console.log(`W/ dataloader: ${diffDataloader} ms`);
    expect(diffDataloader).toBeLessThan(diff);
  });

  it('should fetch all specialties and their sport (Many To One)', async () => {
    server = createApolloServer(orm);
    const {query} = createTestClient(server);
    const res = await query({query: getSpecialtiesQuery});
    expect(res.data).toBeDefined();
    expect(res.errors).toBeUndefined();
    expect(res.data).toMatchSnapshot();
  });

  it('should return the same entities with EntityDataLoader', async () => {
    server = createApolloServer(orm);
    let {query} = createTestClient(server);
    const res = await query({query: getSpecialtiesQuery});
    server = createApolloServer(orm, false);
    query = createTestClient(server).query;
    const resDataloader = await query({query: getSpecialtiesQuery});
    expect(res.data).toBeDefined();
    expect(res.errors).toBeUndefined();
    expect(resDataloader.data).toBeDefined();
    expect(resDataloader.errors).toBeUndefined();
    expect(JSON.stringify(resDataloader.data)).toBe(JSON.stringify(res.data));
  });

  it('should be faster with EntityDataLoader', async () => {
    server = createApolloServer(orm);
    let {query} = createTestClient(server);
    const start = new Date();
    await query({query: getSpecialtiesQuery});
    const end = new Date();
    const diff = end.getTime() - start.getTime();
    server = createApolloServer(orm, false);
    query = createTestClient(server).query;
    const startDataloader = new Date();
    await query({query: getSpecialtiesQuery});
    const endDataloader = new Date();
    const diffDataloader = endDataloader.getTime() - startDataloader.getTime();
    console.log(`W/o dataloader: ${diff} ms`);
    console.log(`W/ dataloader: ${diffDataloader} ms`);
    expect(diffDataloader).toBeLessThan(diff);
  });

  it('should fetch a BIG query', async () => {
    server = createApolloServer(orm);
    const {query} = createTestClient(server);
    const res = await query({query: bigQuery});
    expect(res.data).toBeDefined();
    expect(res.errors).toBeUndefined();
    expect(res.data).toMatchSnapshot();
  });

  it('should return the same entities with EntityDataLoader', async () => {
    server = createApolloServer(orm);
    let {query} = createTestClient(server);
    const res = await query({query: bigQuery});
    server = createApolloServer(orm, false);
    query = createTestClient(server).query;
    const resDataloader = await query({query: bigQuery});
    expect(res.data).toBeDefined();
    expect(res.errors).toBeUndefined();
    expect(resDataloader.data).toBeDefined();
    expect(resDataloader.errors).toBeUndefined();
    expect(JSON.stringify(resDataloader.data)).toBe(JSON.stringify(res.data));
  });

  it('should be faster with EntityDataLoader', async () => {
    server = createApolloServer(orm);
    let {query} = createTestClient(server);
    const start = new Date();
    await query({query: bigQuery});
    const end = new Date();
    const diff = end.getTime() - start.getTime();
    server = createApolloServer(orm, false);
    query = createTestClient(server).query;
    const startDataloader = new Date();
    await query({query: bigQuery});
    const endDataloader = new Date();
    const diffDataloader = endDataloader.getTime() - startDataloader.getTime();
    console.log(`W/o dataloader: ${diff} ms`);
    console.log(`W/ dataloader: ${diffDataloader} ms`);
    expect(diffDataloader).toBeLessThan(diff);
  });

  afterAll(async () => {
    await wipeDatabasePostgreSql(orm.em);
    await dropSchema(orm.em);
    orm.close(true);
  });
});
