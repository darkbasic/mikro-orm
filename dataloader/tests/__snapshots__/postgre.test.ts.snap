// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backend should fetch a BIG query 1`] = `
Object {
  "specialties": Array [
    Object {
      "id": "1",
      "matches": Array [
        Object {
          "date": "2020-03-12T18:00:00.000Z",
          "id": "1",
          "partecipants": Array [
            Object {
              "createdMatches": Array [
                Object {
                  "date": "2020-03-12T18:00:00.000Z",
                  "id": "1",
                },
                Object {
                  "date": "2020-05-20T16:30:00.000Z",
                  "id": "2",
                },
              ],
              "id": "1",
              "name": "Niccol√≤",
              "partecipatedMatches": Array [
                Object {
                  "date": "2020-03-12T18:00:00.000Z",
                  "id": "1",
                },
                Object {
                  "date": "2020-05-20T16:30:00.000Z",
                  "id": "2",
                },
              ],
            },
            Object {
              "createdMatches": Array [],
              "id": "2",
              "name": "Mario",
              "partecipatedMatches": Array [
                Object {
                  "date": "2020-03-12T18:00:00.000Z",
                  "id": "1",
                },
              ],
            },
            Object {
              "createdMatches": Array [],
              "id": "4",
              "name": "Mattia",
              "partecipatedMatches": Array [
                Object {
                  "date": "2020-03-12T18:00:00.000Z",
                  "id": "1",
                },
                Object {
                  "date": "2020-05-20T16:30:00.000Z",
                  "id": "2",
                },
              ],
            },
            Object {
              "createdMatches": Array [],
              "id": "6",
              "name": "Filippo",
              "partecipatedMatches": Array [
                Object {
                  "date": "2020-03-12T18:00:00.000Z",
                  "id": "1",
                },
              ],
            },
          ],
          "site": Object {
            "id": "1",
            "name": "The Beach",
          },
          "specialty": Object {
            "id": "1",
            "name": "2x2 Male",
          },
        },
      ],
      "name": "2x2 Male",
      "sport": Object {
        "id": "1",
        "name": "Beach Volley",
        "sites": Array [
          Object {
            "id": "1",
            "matches": Array [
              Object {
                "date": "2020-03-12T18:00:00.000Z",
                "id": "1",
              },
              Object {
                "date": "2020-05-20T16:30:00.000Z",
                "id": "2",
              },
            ],
            "name": "The Beach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
          Object {
            "id": "2",
            "matches": Array [],
            "name": "Palabeach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
        ],
        "specialties": Array [
          Object {
            "id": "1",
            "name": "2x2 Male",
          },
          Object {
            "id": "7",
            "name": "2x2 Female",
          },
          Object {
            "id": "8",
            "name": "2x2 Mixed",
          },
        ],
      },
    },
    Object {
      "id": "2",
      "matches": Array [],
      "name": "1x1 Male",
      "sport": Object {
        "id": "2",
        "name": "Beach Tennis",
        "sites": Array [
          Object {
            "id": "1",
            "matches": Array [
              Object {
                "date": "2020-03-12T18:00:00.000Z",
                "id": "1",
              },
              Object {
                "date": "2020-05-20T16:30:00.000Z",
                "id": "2",
              },
            ],
            "name": "The Beach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
          Object {
            "id": "2",
            "matches": Array [],
            "name": "Palabeach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
        ],
        "specialties": Array [
          Object {
            "id": "2",
            "name": "1x1 Male",
          },
          Object {
            "id": "3",
            "name": "1x1 Female",
          },
          Object {
            "id": "4",
            "name": "2x2 Male",
          },
          Object {
            "id": "5",
            "name": "2x2 Female",
          },
          Object {
            "id": "6",
            "name": "2x2 Mixed",
          },
        ],
      },
    },
    Object {
      "id": "3",
      "matches": Array [],
      "name": "1x1 Female",
      "sport": Object {
        "id": "2",
        "name": "Beach Tennis",
        "sites": Array [
          Object {
            "id": "1",
            "matches": Array [
              Object {
                "date": "2020-03-12T18:00:00.000Z",
                "id": "1",
              },
              Object {
                "date": "2020-05-20T16:30:00.000Z",
                "id": "2",
              },
            ],
            "name": "The Beach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
          Object {
            "id": "2",
            "matches": Array [],
            "name": "Palabeach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
        ],
        "specialties": Array [
          Object {
            "id": "2",
            "name": "1x1 Male",
          },
          Object {
            "id": "3",
            "name": "1x1 Female",
          },
          Object {
            "id": "4",
            "name": "2x2 Male",
          },
          Object {
            "id": "5",
            "name": "2x2 Female",
          },
          Object {
            "id": "6",
            "name": "2x2 Mixed",
          },
        ],
      },
    },
    Object {
      "id": "4",
      "matches": Array [],
      "name": "2x2 Male",
      "sport": Object {
        "id": "2",
        "name": "Beach Tennis",
        "sites": Array [
          Object {
            "id": "1",
            "matches": Array [
              Object {
                "date": "2020-03-12T18:00:00.000Z",
                "id": "1",
              },
              Object {
                "date": "2020-05-20T16:30:00.000Z",
                "id": "2",
              },
            ],
            "name": "The Beach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
          Object {
            "id": "2",
            "matches": Array [],
            "name": "Palabeach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
        ],
        "specialties": Array [
          Object {
            "id": "2",
            "name": "1x1 Male",
          },
          Object {
            "id": "3",
            "name": "1x1 Female",
          },
          Object {
            "id": "4",
            "name": "2x2 Male",
          },
          Object {
            "id": "5",
            "name": "2x2 Female",
          },
          Object {
            "id": "6",
            "name": "2x2 Mixed",
          },
        ],
      },
    },
    Object {
      "id": "5",
      "matches": Array [],
      "name": "2x2 Female",
      "sport": Object {
        "id": "2",
        "name": "Beach Tennis",
        "sites": Array [
          Object {
            "id": "1",
            "matches": Array [
              Object {
                "date": "2020-03-12T18:00:00.000Z",
                "id": "1",
              },
              Object {
                "date": "2020-05-20T16:30:00.000Z",
                "id": "2",
              },
            ],
            "name": "The Beach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
          Object {
            "id": "2",
            "matches": Array [],
            "name": "Palabeach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
        ],
        "specialties": Array [
          Object {
            "id": "2",
            "name": "1x1 Male",
          },
          Object {
            "id": "3",
            "name": "1x1 Female",
          },
          Object {
            "id": "4",
            "name": "2x2 Male",
          },
          Object {
            "id": "5",
            "name": "2x2 Female",
          },
          Object {
            "id": "6",
            "name": "2x2 Mixed",
          },
        ],
      },
    },
    Object {
      "id": "6",
      "matches": Array [],
      "name": "2x2 Mixed",
      "sport": Object {
        "id": "2",
        "name": "Beach Tennis",
        "sites": Array [
          Object {
            "id": "1",
            "matches": Array [
              Object {
                "date": "2020-03-12T18:00:00.000Z",
                "id": "1",
              },
              Object {
                "date": "2020-05-20T16:30:00.000Z",
                "id": "2",
              },
            ],
            "name": "The Beach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
          Object {
            "id": "2",
            "matches": Array [],
            "name": "Palabeach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
        ],
        "specialties": Array [
          Object {
            "id": "2",
            "name": "1x1 Male",
          },
          Object {
            "id": "3",
            "name": "1x1 Female",
          },
          Object {
            "id": "4",
            "name": "2x2 Male",
          },
          Object {
            "id": "5",
            "name": "2x2 Female",
          },
          Object {
            "id": "6",
            "name": "2x2 Mixed",
          },
        ],
      },
    },
    Object {
      "id": "7",
      "matches": Array [],
      "name": "2x2 Female",
      "sport": Object {
        "id": "1",
        "name": "Beach Volley",
        "sites": Array [
          Object {
            "id": "1",
            "matches": Array [
              Object {
                "date": "2020-03-12T18:00:00.000Z",
                "id": "1",
              },
              Object {
                "date": "2020-05-20T16:30:00.000Z",
                "id": "2",
              },
            ],
            "name": "The Beach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
          Object {
            "id": "2",
            "matches": Array [],
            "name": "Palabeach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
        ],
        "specialties": Array [
          Object {
            "id": "1",
            "name": "2x2 Male",
          },
          Object {
            "id": "7",
            "name": "2x2 Female",
          },
          Object {
            "id": "8",
            "name": "2x2 Mixed",
          },
        ],
      },
    },
    Object {
      "id": "8",
      "matches": Array [
        Object {
          "date": "2020-05-20T16:30:00.000Z",
          "id": "2",
          "partecipants": Array [
            Object {
              "createdMatches": Array [
                Object {
                  "date": "2020-03-12T18:00:00.000Z",
                  "id": "1",
                },
                Object {
                  "date": "2020-05-20T16:30:00.000Z",
                  "id": "2",
                },
              ],
              "id": "1",
              "name": "Niccol√≤",
              "partecipatedMatches": Array [
                Object {
                  "date": "2020-03-12T18:00:00.000Z",
                  "id": "1",
                },
                Object {
                  "date": "2020-05-20T16:30:00.000Z",
                  "id": "2",
                },
              ],
            },
            Object {
              "createdMatches": Array [],
              "id": "4",
              "name": "Mattia",
              "partecipatedMatches": Array [
                Object {
                  "date": "2020-03-12T18:00:00.000Z",
                  "id": "1",
                },
                Object {
                  "date": "2020-05-20T16:30:00.000Z",
                  "id": "2",
                },
              ],
            },
            Object {
              "createdMatches": Array [],
              "id": "7",
              "name": "Sara",
              "partecipatedMatches": Array [
                Object {
                  "date": "2020-05-20T16:30:00.000Z",
                  "id": "2",
                },
              ],
            },
            Object {
              "createdMatches": Array [],
              "id": "9",
              "name": "Giorgia",
              "partecipatedMatches": Array [
                Object {
                  "date": "2020-05-20T16:30:00.000Z",
                  "id": "2",
                },
              ],
            },
          ],
          "site": Object {
            "id": "1",
            "name": "The Beach",
          },
          "specialty": Object {
            "id": "8",
            "name": "2x2 Mixed",
          },
        },
      ],
      "name": "2x2 Mixed",
      "sport": Object {
        "id": "1",
        "name": "Beach Volley",
        "sites": Array [
          Object {
            "id": "1",
            "matches": Array [
              Object {
                "date": "2020-03-12T18:00:00.000Z",
                "id": "1",
              },
              Object {
                "date": "2020-05-20T16:30:00.000Z",
                "id": "2",
              },
            ],
            "name": "The Beach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
          Object {
            "id": "2",
            "matches": Array [],
            "name": "Palabeach",
            "sports": Array [
              Object {
                "id": "1",
                "name": "Beach Volley",
              },
              Object {
                "id": "2",
                "name": "Beach Tennis",
              },
            ],
          },
        ],
        "specialties": Array [
          Object {
            "id": "1",
            "name": "2x2 Male",
          },
          Object {
            "id": "7",
            "name": "2x2 Female",
          },
          Object {
            "id": "8",
            "name": "2x2 Mixed",
          },
        ],
      },
    },
  ],
}
`;

exports[`Backend should fetch all specialties and their sport (Many To One) 1`] = `
Object {
  "specialties": Array [
    Object {
      "id": "1",
      "name": "2x2 Male",
      "sport": Object {
        "id": "1",
        "name": "Beach Volley",
      },
    },
    Object {
      "id": "2",
      "name": "1x1 Male",
      "sport": Object {
        "id": "2",
        "name": "Beach Tennis",
      },
    },
    Object {
      "id": "3",
      "name": "1x1 Female",
      "sport": Object {
        "id": "2",
        "name": "Beach Tennis",
      },
    },
    Object {
      "id": "4",
      "name": "2x2 Male",
      "sport": Object {
        "id": "2",
        "name": "Beach Tennis",
      },
    },
    Object {
      "id": "5",
      "name": "2x2 Female",
      "sport": Object {
        "id": "2",
        "name": "Beach Tennis",
      },
    },
    Object {
      "id": "6",
      "name": "2x2 Mixed",
      "sport": Object {
        "id": "2",
        "name": "Beach Tennis",
      },
    },
    Object {
      "id": "7",
      "name": "2x2 Female",
      "sport": Object {
        "id": "1",
        "name": "Beach Volley",
      },
    },
    Object {
      "id": "8",
      "name": "2x2 Mixed",
      "sport": Object {
        "id": "1",
        "name": "Beach Volley",
      },
    },
  ],
}
`;

exports[`Backend should fetch all sports and their specialties (One To Many) 1`] = `
Object {
  "sports": Array [
    Object {
      "id": "1",
      "name": "Beach Volley",
      "specialties": Array [
        Object {
          "id": "1",
          "name": "2x2 Male",
        },
        Object {
          "id": "7",
          "name": "2x2 Female",
        },
        Object {
          "id": "8",
          "name": "2x2 Mixed",
        },
      ],
    },
    Object {
      "id": "2",
      "name": "Beach Tennis",
      "specialties": Array [
        Object {
          "id": "2",
          "name": "1x1 Male",
        },
        Object {
          "id": "3",
          "name": "1x1 Female",
        },
        Object {
          "id": "4",
          "name": "2x2 Male",
        },
        Object {
          "id": "5",
          "name": "2x2 Female",
        },
        Object {
          "id": "6",
          "name": "2x2 Mixed",
        },
      ],
    },
  ],
}
`;

exports[`Backend should fetch all users 1`] = `
Object {
  "users": Array [
    Object {
      "id": "1",
      "name": "Niccol√≤",
    },
    Object {
      "id": "2",
      "name": "Mario",
    },
    Object {
      "id": "3",
      "name": "Luigi",
    },
    Object {
      "id": "4",
      "name": "Mattia",
    },
    Object {
      "id": "5",
      "name": "Riccardo",
    },
    Object {
      "id": "6",
      "name": "Filippo",
    },
    Object {
      "id": "7",
      "name": "Sara",
    },
    Object {
      "id": "8",
      "name": "Lucia",
    },
    Object {
      "id": "9",
      "name": "Giorgia",
    },
    Object {
      "id": "10",
      "name": "Claudia",
    },
  ],
}
`;
